compiler/modules/nback/scaffold.bal

```diff
@@ -470,7 +470,7 @@ function semTypeRepr(t:SemType ty) returns Repr {
     if w == t:NEVER {
         panic err:impossible("allocate register with never type");
     }
-    int supported = t:NIL|t:BOOLEAN|t:INT|t:FLOAT|t:STRING|t:LIST|t:MAPPING|t:ERROR;
+    int supported = t:NIL|t:BOOLEAN|t:INT|t:FLOAT|t:DECIMAL|t:STRING|t:LIST|t:MAPPING|t:XML|t:TABLE|t:ERROR;
```

Comment:
@jclark I am not sure whether this is the correct way to support any data (In the original form it will cause the panic "unimplemented type")

Reply:
@jclark adding table here (`if (w | supported | t:DECIMAL| t:TABLE) == t:TOP || (w & supported) == w {`) had no effect because w (`1111000000011110111111`) already has all the bits of t:Table (`100000000000000010000`)

Reply:
So why is it getting an unimplemented type error then? (t:DECIMAL and t:TABLE need **not** to be in supported).

What you have here is not right.

Reply:
If I understand this correctly left side (currently (` w | supported | t:DECIMAL) == t:TOP`)) works by adding every thing that is not w or support (because t:TOP is `11111111111111111111111`)). So in the current case this means if the only missing part in the union of w and support is decimal this is true. If this is the case fallowing are the offsets needed here
```
 UT_OBJECT_RO  = 0x06;
 UT_FUNCTION   = 0x0C;
 UT_TYPEDESC   = 0x0D;
 UT_HANDLE     = 0x0E;
 0x0F; // I couldn't find any thing corresponding to this offset
 UT_FUTURE     = 0x10;
 UT_STREAM     = 0x11;
 UT_OBJECT_RW  = 0x16;
```

Reply:
The first `==` is handling the case where the user specified `any`. The `|` supported is dealing with narrowing from the supported types. The `|t:DECIMAL` is dealing with the fact the `<int>any` will have type `any - (float|decimal)`.

Reply:
I just added t:NON_BEHAVIORAL for us to use here. I think this should do it:

```
int maximized= w | supported | t:DECIMAL;
if maximized== t:TOP || maximized == t:NON_BEHAVIOURAL|t:ERROR ...
```

Reply:
If it gets much more complicated, we should just get rid of this checking and return the TaggedRepr. (We should keep the test for NEVER at the beginning).

Reply:
fixed